src/Adapter/Language/CommandHandler/AbstractLanguageHandler.php

```diff
@@ -0,0 +1,205 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Language\CommandHandler;
+
+use Context;
+use Db;
+use ImageManager;
+use ImageType;
+use Language;
+use PrestaShop\PrestaShop\Core\Domain\Language\Exception\CopyingNoPictureException;
+use PrestaShop\PrestaShop\Core\Domain\Language\Exception\LanguageImageUploadingException;
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\IsoCode;
+use Shop;
+
+/**
+ * Encapsulates common legacy behavior for adding/editing language
+ */
+abstract class AbstractLanguageHandler
+{
+    /**
+     * Copies "No picture" image for specific language
+     *
+     * @param IsoCode $isoCode
+     * @param string $noPictureImagePath
+     */
+    protected function copyNoPictureImage(IsoCode $isoCode, $noPictureImagePath)
+    {
+        if (!($temporaryImage = tempnam(_PS_TMP_IMG_DIR_, 'PS'))
+            || !move_uploaded_file($noPictureImagePath, $temporaryImage)
+        ) {
+            return;
+        }
+
+        if (!ImageManager::resize($temporaryImage, _PS_IMG_DIR_ . 'p/' . $isoCode->getValue() . '.jpg')) {
```

Comment:
@eternoendless While migrating the pages, we find out how much "image processing and storage" logic is scattered all around the software. One day, we should write a ImageManager component able to handle this smoothly rather than hardcoding image names and paths